platform/security/security-jaas-ldap/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java

```diff
@@ -457,6 +460,17 @@ private void installEncryptionService() {
     }
   }
 
+  private void installSecurityLogger() {
+
+    BundleContext bundleContext = getContext();
+    if (bundleContext != null) {
+      ServiceReference serviceReference =
+          bundleContext.getServiceReference(SecurityLogger.class.getName());
+      securityLogger = (SecurityLogger) bundleContext.getService(serviceReference);
```

Comment:
Then we're still passing an implementation in the api, which is what I was trying to avoid.

Reply:
You'd still have the api/impl split though so you can hide the implementation dependencies from clients. As described it would make the API OSGI-specific, but you could probably make that pluggable too.

Just a thought, not a suggestion for this PR.

Reply:
I'm going to leave it as is for now.